src/workerd/util/sqlite.c++

```diff
@@ -1096,6 +1104,8 @@ void SqliteDatabase::Query::nextRow() {
   db.currentRegulator = regulator;
 
   int err = sqlite3_step(statement);
+  rowsRead += getRowsRead();
```

Comment:
`getRowsRead()` is culmulative, so I think you want to use `=`, not `+=`.  Same for the write case below.

You should be able to detect this in sqlite-test.c++ by having a single statement that returns multiple rows.

Reply:
You are right, will fix and add a test